tools/kmodules/lisa-in-tree/fetch_lisa_module.py

```diff
@@ -0,0 +1,84 @@
+#! /usr/bin/env python3
+
+#
+# Usage:
+#
+# Method 1 - LISA_HOME
+#
+# If the script finds the LISA_HOME environment variable (present after sourcing the Lisa shell)
+# it will symlink the Lisa module sources from the local Lisa git checkout into the kernel tree.
+# That way updating the local Lisa tree will automatically update the module sources in the kernel
+# tree and the changes will be present after the next kernel rebuild.
+# In this case, the script only needs to be run once.
+#
+# Method 2 - Sparse checkout
+#
+# If the script doesn't find LISA_HOME (e.g. building on an external server without a Lisa checkout)
+# the alternative mode is to directly clone the module sources into the tree. Assuming the paths are
+# set correctly and the remote repo exists:
+# $ ./fetch_lisa_module.py
+# will checkout and symlink the repo accordingly.
+# After the first checkout, re-running the script with a specific git hash will
+# run 'git checkout'.
+# $ ./fetch_lisa_module.py <HASH>
+# It is also possible to pass '-u' in order to fetch the latest sources prior
+# to the checkout.
+# $ ./fetch_lisa_module.py -u <HASH>
+
+import os
+import sys
+import argparse
+import subprocess
+
+# module directory setup
+LISA_MODULE_DIR = "lisa"
+LISA_MODULE_DIR_KERNEL_PATH = "private/gs-google/drivers/soc/arm/vh/kernel"
+
+# sparse clone remote setup
+LISA_MODULE_REMOTE = "git@github.com:mrkajetanp/lisa.git"
+LISA_MODULE_REMOTE_BRANCH = "kmod-in-tree"
+LISA_MODULE_GIT_DIR = "lisa-module-git"
+
+parser = argparse.ArgumentParser(prog='Lisa module in-tree helper')
+init_group = parser.add_mutually_exclusive_group()
+init_group.add_argument('hash', nargs='?', help='Lisa repository hash to checkout')
+parser.add_argument('-u', '--update', action='store_true', help='Run git fetch before checking out the hash')
+args = parser.parse_args()
+
+# LISA_HOME is present, use the bundled module
+module_kernel_path = os.path.join('.', LISA_MODULE_DIR_KERNEL_PATH, LISA_MODULE_DIR)
+
+if 'LISA_HOME' in os.environ:
+    LISA_HOME = os.environ['LISA_HOME']
+    lisa_home_module_src = os.path.join(LISA_HOME, 'lisa/_assets/kmodules/lisa')
+    link_cmd = [
+        'ln', '-sr', lisa_home_module_src, module_kernel_path
+    ]
+    subprocess.check_call(link_cmd)
```

Comment:
Can use [os.symlink](https://docs.python.org/3/library/os.html#os.symlink) or [Path.symlink_to](https://docs.python.org/3/library/pathlib.html#pathlib.Path.symlink_to) instead of relying on a shell command